toothpick-runtime/src/main/java/toothpick/InjectionCycleFormatter.java

```diff
@@ -0,0 +1,72 @@
+package toothpick;
+
+import java.util.List;
+
+/**
+ * Formats an injection cycle to its String representation.
+ */
+public class InjectionCycleFormatter {
```

Comment:
make it non static, and I would love to see a few tests to see what the messages look like.